src/main/java/com/ericsson/ei/subscriptionhandler/InformSubscriber.java

```diff
@@ -0,0 +1,429 @@
+
+        String subject = getSubscriptionField("emailSubject", subscriptionJson);
+
+        MultiValueMap<String, String> mapNotificationMessage = mapNotificationMessage(aggregatedObject,
+                subscriptionJson);
+
+        if (notificationType.trim().equals("REST_POST")) {
+            LOGGER.debug("Notification through REST_POST");
+            boolean success = false;
+
+            if (notificationMeta.contains("?")) {
+                LOGGER.debug("Unformatted notificationMeta = " + notificationMeta);
+                try {
+                    notificationMeta = replaceParamsValuesWithAggregatedData(aggregatedObject, notificationMeta);
+                    LOGGER.debug("Formatted notificationMeta = " + notificationMeta);
+                } catch (URISyntaxException e) {
+                    LOGGER.error("Failed to reformat url reason: " + e.getMessage());
+                }
+            }
+
+            // Prepare request headers
+            HttpHeaders headers = new HttpHeaders();
+            headers = prepareHeaders(headers, notificationMeta, subscriptionJson);
+
+            success = restTemplate.postDataMultiValue(notificationMeta, mapNotificationMessage, headers);
+
+            for (int i = 0; i < failAttempt; i++) {
+                if (success) {
+                    break;
+                }
+                success = restTemplate.postDataMultiValue(notificationMeta, mapNotificationMessage, headers);
+                LOGGER.debug("After retrying for " + (i + 1) + " times, the result is : " + success);
+            }
+
+            if (!success) {
+                saveMissedNotificationToDB(aggregatedObject, subscriptionName, notificationMeta);
+            }
+        }
+
+        if (notificationType.trim().equals("MAIL")) {
+            LOGGER.debug("Notification through EMAIL");
+            try {
+                sendMail.sendMail(notificationMeta, String.valueOf((mapNotificationMessage.get("")).get(0)), subject);
+            } catch (MessagingException e) {
+                e.printStackTrace();
+                LOGGER.error(e.getMessage());
+            }
+        }
+
+    }
+
+    /**
+     * This method prepares headers to be used when making a rest call with the
+     * method POST.
+     *
+     * @param headers
+     * @param notificationMeta
+     * @param subscriptionJson
+     * @return
+     */
+    private HttpHeaders prepareHeaders(HttpHeaders headers, String notificationMeta, JsonNode subscriptionJson) {
+        // Setting Content Type
+        String headerContentMediaType = getSubscriptionField("restPostBodyMediaType", subscriptionJson);
+        headers.setContentType(MediaType.valueOf(headerContentMediaType));
+        LOGGER.debug("Successfully added header: "
+                + String.format("'%s':'%s'", "restPostBodyMediaType", headerContentMediaType));
+
+        // Adding authentication if any
+        String authType = getSubscriptionField("authenticationType", subscriptionJson);
```

Comment:
move this check into the method. Just call the method here

Reply:
Done.